File: frontend/src/types/domain/company.ts
import { Address } from './address';

export interface Company {
    id: string; // UUID
    name: string;
    taxId: string;
    address: Address;
    email?: string;
    phoneNumber?: string;
}

export interface CompanyTableData {
    id: string;
    lp: string;
    name: string;
    taxId: string;
    email: string;
    phone: string;
    address: string;
}

File: backend/MasApi/Models/Company.cs
using System.ComponentModel.DataAnnotations;

namespace MasApi.Models;

public class Company: BaseModel
{
    [Key]
    public required Guid Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public required Address Address { get; set; }
    public required string TaxId { get; set; }
    public string? Email { get; set; }
    public string? PhoneNumber { get; set; }

    public ICollection<Invoice>? Invoices { get; set; }
}
